src/app/features/container/utils/route-helpers.ts

```diff
@@ -0,0 +1,80 @@
+import { RouteUrls } from '@shared/route-urls';
+
+import { isKnownPopupRoute } from './get-popup-header';
+
+function isHomePage(pathname: RouteUrls) {
```

Comment:
@pete-watters I was thinking about this implementation yesterday where you are using variants of one container with lots of detecting of routes, I am wondering what you think about an alt approach where we just have several, more specific, containers that wrap the routes that need it? I know the code might be less DRY, but sometimes it makes scaling the code a bit easier long term? Same question for headers?

Reply:
I agree, its a real mess. A lot of this complication came from having different background colours for different page types e.g. `Home` and `Page`. It was also a bridge to get us away from how we were doing things before - storing `Header` components in application state. 

We are getting rid of the colour variations now and having more normalised page backgrounds so when I am working on that I will try and simplify everything and ditch these complex variants. 

Once everything is working and signed off we can assess and figure out a better pattern to use